compiler/rustc_codegen_ssa/src/back/link/raw_dylib.rs

```diff
@@ -230,40 +230,66 @@ fn create_elf_raw_dylib_stub(sess: &Session, soname: &str, symbols: &[DllImport]
         Endian::Little => object::Endianness::Little,
         Endian::Big => object::Endianness::Big,
     };
+
     let mut stub = write::Writer::new(endianness, true, &mut stub_buf);
 
+    let mut vers = Vec::new();
+    let mut vers_map = FxHashMap::default();
+    let mut syms = Vec::new();
+
+    for symbol in symbols {
+        let symbol_name = symbol.name.as_str();
+        if let Some((name, version_name)) = symbol_name.split_once('@') {
+            assert!(!version_name.contains('@'));
+            let dynstr = stub.add_dynamic_string(name.as_bytes());
+            let ver = if let Some(&ver_id) = vers_map.get(version_name) {
+                ver_id
+            } else {
+                let id = vers.len();
+                vers_map.insert(version_name, id);
+                let dynstr = stub.add_dynamic_string(version_name.as_bytes());
+                vers.push((version_name, dynstr));
```

Comment:
Is the `version_name` inside `vers` used anywhere?

Reply:
Not yet. The existing code does the same thing for symbol names, so I think it might be useful for version names, too.